pkg/cloud/services/networking/securitygroups.go

```diff
@@ -200,7 +201,15 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 	workerRules := append([]resolvedSecurityGroupRuleSpec{}, defaultRules...)
 
 	controlPlaneRules = append(controlPlaneRules, getSGControlPlaneHTTPS()...)
-	workerRules = append(workerRules, getSGWorkerNodePort()...)
+
+	// Fetch subnet to use for worker node port rules
+	// TODO: add case where subnets are used. However need a way of resolving the subnet to get it's CIDR
+	if openStackCluster.Spec.ManagedSubnets != nil && len(openStackCluster.Spec.ManagedSubnets) > 0 {
+		SubnetCIDR = openStackCluster.Spec.ManagedSubnets[0].CIDR
+	} else {
+		SubnetCIDR = "0.0.0.0/0"
+	}
```

Comment:
Can we not use remote group for this? This won't work on deployments which aren't using managed subnets.

Reply:
I just changed the behaviour to make use of status.Network.Subnet instead to support use cases where bring your own network/subnet is used.

I do not think we could use remote group due to load balancers talking over nodeports?
Load Balancers are created under the "service" project in default openstack meaning it is not added to any security groups inside the project where the cluster is running.

Reply:
I also left a question in the code that may be good to bring up here.

If no IPv4 subnet is found the code currently defaults to `0.0.0.0/0` to preserve current behaviour.

We could however return an error if no IPv4 subnet is found or potentially fallback to just allowing nodes by using remote groups (this will however break octavia load balancers created using the cloud provider which feels kinda bad).

Reply:
In other places we just add the rule twice with 2 remote groups, one for the control plane and one for workers. This covers machines even when they aren't using the default cluster network. Is there a reason we can't do that here?

I'd like to reduce our reliance on the cluster's status.Network over time. There are new, interesting deployments that it doesn't play nicely with, e.g. external control planes and spine/leaf.

Reply:
> In other places we just add the rule twice with 2 remote groups, one for the control plane and one for workers. This covers machines even when they aren't using the default cluster network. Is there a reason we can't do that here?
> 
> I'd like to reduce our reliance on the cluster's status.Network over time. There are new, interesting deployments that it doesn't play nicely with, e.g. external control planes and spine/leaf.

It would break load balancers deployed using the cloud provider (octavia) since they are not in any security group we control (know the ID of). I think it is a really common use case and support out of the box would be nice

Reply:
Just wanted to follow up on this.

We could add security groups meaning all cluster nodes can contact each other and it will make the conformance test pass. However it would cause LBs created with the cloud provider to fail. I'm not at this point of any way of allowing LBs outside allowing IPs directly towards nodes. 

I also think having a default where LBs "work" is preferable to have a default that works "out of the box".